src/Emulator/Visuals/Visuals.js

```diff
@@ -1,11 +1,169 @@
+      });
+  };
+
+  // Handles what to do when starting drag and stopping drag
+  const onStart = () => {
+    setDrags( drags => drags + 1);
+  };
+
+  const onStop = () => {
+    setDrags( drags => drags - 1);
+  };
+
+  // When image is added, adjusts the container and drag handler sizes to accomodate for image size
+  const imageLoad = () => {
+    const additionalSpace = 50;
+    setContainerDimensions({
+      height: imageRef.current.offsetHeight + containerHeight,
+      width: imageRef.current.offsetWidth + containerWidth,
+    });
+    setContentDimensions({
+      height: imageRef.current.offsetHeight + additionalSpace,
+      width: imageRef.current.offsetWidth + additionalSpace,
+    });
+    setSrc(image);
+    return {containerDimensions, contentDimensions};
+    };
+
+  // Zoom in function and adjusts drag handler size (the invisible space you can click and start drag with)
+  function plusClick() {
+    const zoomAmount = .1;
+    const additionalContentSpace = 20;
+      setZoom(zoom => zoom + zoomAmount);
+      setContentDimensions({
+          height: contentDimensions.height + additionalContentSpace,
+          width: contentDimensions.width + additionalContentSpace,
+        });
+  }
+
+  // Zoom out function and adjusts drag handler size (the invisible space you can click and start drag with)
+  function minusClick() {
+    const zoomAmount = .1;
+    const additionalContentSpace = 20;
+    setZoom(zoom => zoom - zoomAmount);
+    setContentDimensions({
+      height: contentDimensions.height - additionalContentSpace,
+      width: contentDimensions.width - additionalContentSpace,
+    });
+  }
+
+  // Full screen function; changes element resolution to container's max dimensions when when maxRes and toggleMax state changes
+  useEffect(() => {
+    if (toggleMax === true) {
+      setContentDimensions({
+        height: maxRes.height,
+        width: maxRes.width,
+      });
+    } else {
+      setContentDimensions({
+        height: maxRes.height,
+        width: maxRes.width,
+      });
+    }
+    }, [maxRes, toggleMax]);
+
+  // Full screen function; finds cotainer and container space resolution sizes and sets them so that image fills the whole container using true/false states
+  function maxClick() {
+    setMax(!toggleMax);
+    if (toggleMax === false) {
+      setMaxRes({
+        height: containerDimensions.height,
+        width: containerDimensions.width,
+        prevHeight: contentDimensions.height,
+        prevWidth: contentDimensions.width,
+      });
+      setPos({
+        x: 0, y: 0,
+      });
+    } else {
+      const centerX = 475;
+      const centerY = 475;
+      setMaxRes({
+        height: maxRes.prevHeight,
+        width: maxRes.prevHeight,
+        prevHeight: maxRes.height,
+        prevWidth: maxRes.width,
+      });
+      setPos({
+        x: centerX, y: centerY,
+      });
+    }
+  }
+
+  function hideClick() {
+      return (
+          setHideImage(!hideImage)
+      );
+  }
```

Comment:
One interesting thing I found out about react state hooks is that the state can be boolean, so when I wrote `setHideImage(!hideImage)` I was trying to inverse the states of true and false. The alternative way I could've done it instead is to move `setHideImage(false)` below the function and delete the `return` line but I thought it looked much fancier to return it instead since it still did the job of hiding/revealing images as intended. Was the alternative method what I should have done instead?

Reply:
Hm... I think you can just do `setHideImage(!hideImage)` or even `setHideImage(prevHideImage => !prevHideImage)` because this function doesn't actually return anything it might be a little confusing to someone looking back at it later so I would remove the return if you can 😅